### PR TITLE
Upgrade rocketmq for pubsub

### PR DESCRIPTION
**Pull request by zach030**
# Description

Upgrade rocketmq client with apache/rocketmq-client-go/v2 and remove dependence on third-party client.

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by daixiang0** [comment]
Could share why need to upgrade?

**Comment by CodeMonkeyLeet** [comment]
@zach030 can you also look at the build failures? It looks like the resulting go.sum is incomplete so a `go mod tidy` is needed

**Comment by zach030** [comment]
> Could share why need to upgrade?

The previous pubsub component rocketmq depends on third-party libraries. According to the opinions of the RocketMQ community, I remove it and make it purely base on apache/rocketmq-client-go

File: pubsub/rocketmq/rocketmq.go [hunk 1]
```diff
@@ -1,301 +1,306 @@
-// ------------------------------------------------------------
```
Comment (CodeMonkeyLeet) [inline pubsub/rocketmq/rocketmq.go:1]: Please do not remove the licensing and copyright headers

File: pubsub/rocketmq/metadata.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
+package rocketmq
```
Comment (CodeMonkeyLeet) [inline pubsub/rocketmq/metadata.go:1]: Please add the standard copyright and licensing headers to source code.

File: tests/config/pubsub/rocketmq/pubsub.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  version: v1
+  metadata:
+    - name: nameServer
+      value: "127.0.0.1:9876"
+    - name: consumerGroup
+      value: "dapr.rocketmq.producer"
+    - name: accessKey
+      value: "RocketMQ"
+    - name: secretKey
+      value: "12345"
```
Comment (CodeMonkeyLeet) [inline tests/config/pubsub/rocketmq/pubsub.yml:16]: Nit: Missing newline at EOF for linux text files.

**Comment by CodeMonkeyLeet** [comment]
@yaron2 @artursouza This is a fairly significant change, have the maintainers approved this? It looks in line with previous requests such as #890

**Comment by CodeMonkeyLeet** [comment]
@zach030 Still hitting build failures with the go.sum issues

**Comment by beiwei30** [comment]
@zach030 would you mind to resolve the conflict and make CI pass? I plan to review ur change in this week :)

**Comment by zach030** [comment]
> @zach030 would you mind to resolve the conflict and make CI pass? I plan to review ur change in this week :)

Ok I will resolve conflict!

**Comment by beiwei30** [comment]
@zach030 pls. fix the unit test :)

> FAIL	github.com/dapr/components-contrib/pubsub/rocketmq	0.176s

**Comment by zach030** [comment]
> @zach030 pls. fix the unit test :)
> 
> > FAIL	github.com/dapr/components-contrib/pubsub/rocketmq	0.176s

already fixed!

**Comment by sduo** [comment]
Is any possible merge this branch in to 1.6

**Comment by yaron2** [comment]
@zach030 please resolve the conflicts and we can then merge if @beiwei30 is happy with the changes.

**Comment by yaron2** [comment]
> Is any possible merge this branch in to 1.6

We can put this into 1.6 as long as there's a corresponding docs issue in place.
/cc @zach030 @x-shadow-man

**Comment by sduo** [comment]
> > Is any possible merge this branch in to 1.6
> 
> We can put this into 1.6 as long as there's a corresponding docs issue in place. /cc @zach030 @x-shadow-man

Thanks for help, :)

**Comment by sduo** [comment]
@zach030 please help

**Comment by zach030** [comment]
> 



> @zach030 please help

ok, I will complete it soon!

**Comment by sduo** [comment]
> > 
> 
> > @zach030 please help
> 
> ok, I will complete it soon!

thanks ~I‘m not good at Go, so thanks ~

**Comment by zach030** [comment]
@yaron2 please run the workflow

**Comment by zach030** [comment]
thanks!

> @yaron2 please run the workflow

@yaron2 I wondered how to run DCO checks and how to write the corresponding doc, any example for me? thanks for your patience :)

**Comment by zach030** [comment]
@yaron2 I find it's quite confused for me to fix conflict while rebasing. So can I checkout a new branch and open another PR?

**Comment by yaron2** [comment]
> @yaron2 I find it's quite confused for me to fix conflict while rebasing. So can I checkout a new branch and open another PR?

Yep go for it

**Comment by zach030** [comment]
new pr #1383

**Pull request closed without merging**